types/resolve/resolve-tests.ts

```diff
@@ -100,8 +100,13 @@ function test_options_sync() {
     });
     console.log(resolved);
     resolved = resolve.sync('typescript', {
-        readFileSync(file, charset) {
-            return fs.readFileSync(file, charset);
+        readFileSync(file) {
+            return fs.readFileSync(file);
+        },
+    });
+    resolved = resolve.sync('typescript', {
+        readPackageSync(readFileSync, file) {
+            return JSON.parse(readFileSync(file));
```

Comment:
`JSON.parse` in node can take a `Buffer`, so the type error is wrong. Should I just cast it?

```sh-session
$ node -p 'JSON.parse(Buffer.from(JSON.stringify({ works: true})))'
{ works: true }
```

/cc @orta

Reply:
`JSON.parse` only takes in a `Buffer` because it does `.toString()` on the input (https://tc39.es/ecma262/#sec-json.parse, step 1). `JSON.parse(1)` and `JSON.parse(null)` also "work", but that doesn't mean we should widen the input to accept anything. I'd recommend being more explicit (i.e., `readFileSync(file, "utf8")` or `JSON.parse(readFileSync(file).toString("utf8"))`, etc.

Reply:
Either way, the cast is fine in this case since we're testing the types, not the runtime behavior.

Reply:
If it takes any toStringable, then the proper type is `string | { toString: () => string }`, no?

Reply:
> If it takes any toStringable, then the proper type is `string | { toString: () => string }`, no?

I went for that, with the nice side effect of removing the dependency on `@types/node`

[`3e4ee0a` (#51182)](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51182/commits/3e4ee0ae39e4bd66fb5ef040dd12d29f823a0e4c)